release.py

```diff
@@ -46,10 +46,16 @@ def workspace_is_clean():
                          stderr=subprocess.STDOUT)
     return len(p.stdout.readlines()) == 0
 
-def is_draft_release(version_tag):
-    g = Github()
-    release = g.get_repo("buildbuddy-io/buildbuddy").get_release(version_tag)
-    return release.draft
+def is_draft_release(github_token, version_tag):
+    query_url = f"https://api.github.com/repos/buildbuddy-io/buildbuddy/releases/tags/{version_tag}"
+    headers = {'Authorization': f'token {github_token}'}
+    r = requests.get(query_url, headers=headers)
+    if r.status_code == 401:
+        die("Invalid github credentials. Did you pass a valid github token?")
+    elif r.status_code == 404:
+        die(f"No release found for version tag {version_tag}.")
```

Comment:
I think we can return false in this case? (no release found means no draft release was found either)

Reply:
Refactored so that if no release is found, we also skip the version bump. I thought this could be helpful if we create a tag with release.py, but the workflow that creates the release fails. We may want to rerun release.py, but wouldn't expect it to bump the version again